docs/custom-domain/README.md

```diff
@@ -1,56 +1,96 @@
+## Prerequisites
+- [Azure CLI](https://docs.microsoft.com/cli/azure/?view=azure-cli-latest): version 2.4.0 or higher
+  - Consider using [Azure Cloud Shell](https://docs.microsoft.com/azure/cloud-shell/overview)
+- A _premium_ Azure Container Registry. See [here](https://docs.microsoft.com/azure/container-registry/container-registry-get-started-azure-cli) for instructions on how to create one.
+- Your custom domain names. The following two are required:
+  - Custom registry domain to access the registry REST API. Example for the `contoso.com` domain: `container-registry.contoso.com` 
+  - Custom data domain to access the registry content. Again, example for `contoso.com`: `eastus-registry-data.contoso.com`
+    - Note that the custom data domain is region specific. For geo-replicated registries, each region should have its own custom data endpoint.
+
+  For each domain, you must prepare a single PEM formatted file containing the TLS private key and public certificate:
+  
+  ```
+  -----BEGIN PRIVATE KEY-----  
+  .....  
+  -----END PRIVATE KEY-----  
+  -----BEGIN CERTIFICATE-----  
+  .....  
+  -----END CERTIFICATE-----
+  ```
+  
+  For example, using [openssl](https://github.com/openssl/openssl):
+  - Create a self-signed public cert and private key
+     - `openssl req -nodes -x509 -newkey rsa:4096 -keyout container-registry.contoso.com.key.pem -out container-registry.contoso.com.cert.pem -days 365 -subj '/CN=container-registry.contoso.com/O=Contoso./C=US'`
+  - Create a single file containing both the public certificate and private key
+     - `cat container-registry.contoso.com.key.pem >> container-registry-contoso-com-pem`
+     - `cat container-registry.contoso.com.cert.pem >> container-registry-contoso-com-pem`
+  - For each data domain, follow the same steps above to prepare the PEM formatted files containing the public certificate and private key.
+  
+  Azure Key Vault allows you to [create](https://docs.microsoft.com/azure/key-vault/certificate-scenarios) Certificate Authority (CA) signed certificates. 
+  - If you choose to use the Azure Portal to create the certificates, be sure to select certificate content type as PEM.
+ 
+## Prepare your existing registry
+We will enable two features on your registry:
+- Data Endpoints:\
+  This feature provides a dedicated endpoint for downloading content from your registry. If you have a registry in East US, on enabling this feature, a data endpoint is automatically created for you: `myregistry.eastus.data.azurecr.io`
+  
+- ACR Managed Identities:\
+  Managed Identities provide a mechanism to associate an Azure Active Directory identity with your registry, while relieving you of the burden of managing credentials. To learn more, see the documentation [here](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview).\
+ ACR supports both user assigned and system assigned managed identities.
+
+### Enable preview features
```

Comment:
Maybe reword heading: "Enable data endpoints and managed identities"

Reply:
Done